### PR TITLE
chore(websocket): Replace golang.org/x/net/websocket to github.com/gorilla/websocket

### LINKED ISSUE #751: Replace `golang.org/x/net/websocket` to `github.com/gorilla/websocket`
`golang.org/x/net/websocket` is deprecated. As specified in its doc: 

> This package currently lacks some features found in alternative and more actively maintained WebSocket packages

Kind of related to https://github.com/Scalingo/go-scalingo/issues/219. It could be nice to handle both at once.

### PR DESCRIPTION
**Pull request by QuentinEscudierScalingo**
~~Waiting for the release of go-scalingo to update vendors~~

- [x] Add a changelog entry in the section "To Be Released" of CHANGELOG.md

Fix  #751

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### To be Released
 
 * deps(urfave/cli) Bumpd from 2.17.1 to 2.20.2: fix options display regression in --help [#814](https://github.com/Scalingo/cli/pull/814)
+* deps(websocket) Replace golang.org/x/net/websocket to github.com/gorilla/websocket [#822](https://github.com/Scalingo/cli/pull/822)
```
Comment (EtienneM) [inline CHANGELOG.md:6]: nitpick: English wording
```suggestion
* deps(websocket) Replace golang.org/x/net/websocket with github.com/gorilla/websocket [#822](https://github.com/Scalingo/cli/pull/822)
```

;)

File: logs/logs.go [hunk 2]
```diff
@@ -155,13 +159,16 @@ func Stream(logsRawURL string, filter string) error {
...
-		err := websocket.JSON.Receive(conn, &event)
+		err := conn.ReadJSON(&event)
 		if err != nil {
 			conn.Close()
 			if err == stdio.EOF {
 				debug.Println("Remote server broke the connection, reconnecting")
 				for err != nil {
-					conn, err = websocket.Dial(logsURLString, "", "http://scalingo-cli.local/"+config.Version)
+					header := http.Header{}
+					header.Add("Origin", fmt.Sprintf("http://scalingo-cli.local/%s", config.Version))
```
Comment (Soulou) [inline logs/logs.go:169]: Why not reusing the previous header?

**Review by Soulou — CHANGES_REQUESTED** [review]

File: logs/logs.go [hunk 3]
```diff
@@ -136,10 +137,13 @@ func Stream(logsRawURL string, filter string) error {
...
 	}
 
-	conn, err := websocket.Dial(logsURLString, "", "http://scalingo-cli.local/"+config.Version)
+	header := http.Header{}
+	header.Add("Origin", fmt.Sprintf("http://scalingo-cli.local/%s", config.Version))
+	conn, resp, err := websocket.DefaultDialer.DialContext(ctx, logsURLString, header)
 	if err != nil {
 		return errgo.Mask(err, errgo.Any)
 	}
+	defer resp.Body.Close()
```
[the remark this answers is not in the document]
Comment (QuentinEscudierScalingo) [reply logs/logs.go:146]: Sorry I totally forgot but if the resp.Body isn't closed, the linter throws an error `response body must be closed (bodyclose)go-golangci-lint`. That's in part why I close de `resp.Body`.

By the way I agree with the conn, no need to close it

**Comment by QuentinEscudierScalingo** [comment]
> 

As the linter disagree with the `no resp.Body close`, we have the solution to skip the linter of just to keep the resp.Body.Close and I think it will be more appropriate to keep the resp.Body.Close

**Comment by Soulou** [comment]
Let's keep both in this case

**Pull request merged**